internal/httpproxy/cert_reloader.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Twingate Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package httpproxy
+
+import (
+	"crypto/tls"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"go.uber.org/zap"
+)
+
+type certReloader struct {
+	mu       sync.RWMutex
+	certFile string
+	keyFile  string
+	watching chan bool
+	cert     *tls.Certificate
+	logger   *zap.SugaredLogger
+}
+
+func newCertReloader(certFile, keyFile string, logger *zap.SugaredLogger) *certReloader {
+	return &certReloader{
+		certFile: certFile,
+		keyFile:  keyFile,
+		logger:   logger,
+		watching: make(chan bool),
+	}
+}
+
+func (cr *certReloader) load() error {
+	cert, err := tls.LoadX509KeyPair(cr.certFile, cr.keyFile)
+	if err != nil {
+		return err
+	}
+
+	cr.mu.Lock()
+	defer cr.mu.Unlock()
+
+	cr.cert = &cert
+	cr.logger.Info("loaded cert and key files")
```

Comment:
log here probably not needed (since you're not logging the error case here) - let the caller decide how/what to log?